torch/accelerator/__init__.py

```diff
@@ -43,35 +46,47 @@ def is_available() -> bool:
 
         >>> assert torch.accelerator.is_available() "No available accelerators detected."
     """
-    return device_count() > 0
+    # Why not just check "device_count() > 0" like other is_available call?
+    # Because device like CUDA have a python implementation of is_available that is
+    # non-poisoning and some features like Dataloader rely on it.
+    # So we are careful to delegate to the Python version of the accelerator here
+    acc = current_accelerator()
```

Comment:
Is it necessary to call `torch._C._accelerator_getAccelerator()` here to decouple with `current_accelerator` and avoid reentering`current_accelerator` if the user call `current_accelerator(check_available=True)`